core/app/models/spree/payment.rb

```diff
@@ -165,9 +165,14 @@ def profiles_supported?
         payment_method.respond_to?(:payment_profiles_supported?) && payment_method.payment_profiles_supported?
       end
 
+      # Update from upstream 3e0b405a
       def create_payment_profile
-        return unless source.respond_to?(:has_payment_profile?) && !source.has_payment_profile? &&
```

Comment:
`has_payment_profile?` isn't really meant for checking if a profile needs to be created ironically enough.